packages/pymedphys_dicom/src/pymedphys_dicom/dicom/_level2/anonymise.py

```diff
@@ -215,7 +215,10 @@ def anonymise_dataset(
 
     ds_anon = _anonymise_tags(ds_anon, keywords_to_anonymise, replace_values)
 
-    return ds_anon
+    if copy_dataset:
+        return ds_anon
+    else:
+        return None
```

Comment:
I don't think this return None is needed. I think the function should just be left to finish...